ykrt/src/deopt.rs

```diff
@@ -255,15 +255,25 @@ unsafe extern "C" fn __ykrt_deopt(
         }
     }
 
-    unsafe { framerec.reconstruct_frames(frameaddr) }
+    // FIXME: This print doesn't really serve a purpose anymore, since `deoptimise` will always be
+    // followed by `exit-jit-code` as there's no other way to leave a trace.
+    #[cfg(feature = "yk_jitstate_debug")]
+    print_jit_state("exit-jit-code");
```

Comment:
So this is never printed?

Surely we don't want that, and it must break lots of tests?

Perhaps the deopt code should print this somehow?

Reply:
This is the deopt code.

I think what Lukas is saying is that the only way for a trace to complete is for a guard to fail so "deoptimise" subsumes "exit-jit-code".

Reply:
What @ltratt said. "deoptimise" will always be followed by "exit-jit-code" so there's no need to have both. I'll update the tests accordingly.

Reply:
As long as we still get a jitstate print when we exit jitted code, I'm happy :)

Reply:
Ah! I get it! Thanks